src/main/java/org/openmaptiles/layers/Housenumber.java

```diff
@@ -59,13 +65,63 @@ public class Housenumber implements
   Tables.OsmHousenumberPoint.Handler,
   ForwardingProfile.FeaturePostProcessor {
 
-  public Housenumber(Translations translations, PlanetilerConfig config, Stats stats) {}
+  private static final Logger LOGGER = LoggerFactory.getLogger(Housenumber.class);
+  private static final Character OSM_SEPARATOR = ';';
+  private static final String DISPLAY_SEPARATOR = "–";
+  private static final Pattern NO_CONVERSION_PATTERN = Pattern.compile("[^0-9;]");
+  private final Stats stats;
+
+  public Housenumber(Translations translations, PlanetilerConfig config, Stats stats) {
+    this.stats = stats;
+  }
+
+  private static String displayHousenumberNonumeric(String[] numbers) {
+    return numbers[0]
+      .concat(DISPLAY_SEPARATOR)
+      .concat(numbers[numbers.length - 1]);
+  }
+
+  protected static String displayHousenumber(String housenumber) {
+    if (housenumber.indexOf(OSM_SEPARATOR) < 0) {
+      return housenumber;
+    }
+
+    String[] numbers = Arrays.stream(housenumber.split(OSM_SEPARATOR.toString()))
+      .filter(n -> !n.trim().isEmpty())
```

Comment:
Do we want to trim the output?

```suggestion
      .map(String::trim)
      .filter(n -> !n.isEmpty())
```